src/services/formatting/smartIndenter.ts

```diff
@@ -174,22 +179,24 @@ namespace ts.formatting {
                     indentationDelta += options.indentSize;
                 }
 
+                // Update `current` and `parent`.
+
+                const useTrueStart =

```

Comment:
You should put some of the explanation you have written-up on this pull request into the code as comments. The reasoning behind some of this code is really hard to follow without understanding the problems it is trying to solve.

Reply:
added.